engine/src/main/java/org/camunda/bpm/engine/context/ProcessEngineContext.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.context;
+
+import org.camunda.bpm.engine.impl.context.ProcessEngineContextImpl;
+
+/**
+ * <p>A utility to declare a new Process Engine Context in order
+ * for database operations to be separated in a new transaction.</p>
```

Comment:
I think we should go into more detail here what a context is, i.e. explain that a new process engine context means a fresh cache and give some details:

* Fetching data will hit the database and will not resolve changes made in an outer command
* The changes will be flushed as soon as the nested command returns (even if the transaction does not commit at that point)

To explain these things, we may also have to establish that this is relevant in case of nested commands. This could for example be done by first explaining the default behavior (for an "outer" command, that the engine always creates a new context; for a nested command, it participates in the outer context), and then explaining that this util can be used to control if a nested command should participate in the existing context or should have its own.

Reply:
I extended the javadocs [here](https://github.com/camunda/camunda-bpm-platform/blob/1ac1325eb03339abb7ab1e5b785b92184b0ddb52/engine/src/main/java/org/camunda/bpm/engine/context/ProcessEngineContext.java#L24-L57). Maybe we can have a follow-up before I port the changes to the docs as well.